### ISSUE TITLE
Notes on tag line breaks for us to keep in mind

### ISSUE THREAD

**Issue by jacob-gg** [uvastatlab/uvastatlab#2]
`drupal()` currently encloses code/output chunks with sets of tags that are all on the same line, e.g.:

```
<div class="rds-article--code" aria-label="code"><pre><code>
some code
</code></pre></div>
```

During Drupal article updating, we've been using split-line tags:

```
<div class="rds-article--code" aria-label="code">
<pre>
<code>
some code
</code>
</pre>
</div>
```

Since we use \<pre\>, that (should) create the desired single line of space before/after chunks, although we're obviously having some white space issues currently.

Hopefully, the Comms/UX CSS updates on the backend shrink the masses of white space we currently have. If so, then we may want to update the function to insert \n between the tags like so:

```
  # opening code block
  p <- gsub(pattern = '<pre class=(\"r\"|\"python\")><code>',
            replacement = '<div class="rds-article--code" aria-label="code">\n<pre>\n<code>\n',
            p)
  
  # opening output block
  # find <pre><code> at start of line
  p <- gsub(pattern = '\n<pre><code>',
            replacement = '\n<div class="rds-article--output" aria-label="output">\n<pre>\n<code>\n',
            p)
  
  # closing code and output blocks
  p <- gsub(pattern = '</code></pre>',
            replacement = '\n</code>\n</pre>\n</div>',
            p)
```

If for some reason Comms/UX _can't_ do the white space fixes on the backend... Then we may need to try and shrink the space ourselves by converting the enclosing tags to all be on one line (eliminating the lines currently conferred by \<pre\>). I imagine I could knock all ~100 articles out in one day if need be: It sounds like it'd just be a fast find-replace job for each one.

**Comment by clayford** [comment]
Yes, good catch! Feel free to update the function. I really hope we don't need to update all 100 articles. Ugh. I know you say you can do it, but I would totally help with that.

**Comment by jacob-gg** [comment]
Ach, we may not even have the ability to make those changes even if we wanted to. I just tested in Drupal what happens if we directly insert the following:

```
<div aria-label="code" class="rds-article--code"><pre><code>
choose(10,2)
</code></pre></div>

<div aria-label="output" class="rds-article--output"><pre><code>
[1] 45
</code></pre></div>
```

And... Drupal rejects the same-line tags and reformats to:

```
<div aria-label="code" class="rds-article--code">
<pre>
<code>
choose(10,2)
</code></pre>
</div>

<div aria-label="output" class="rds-article--output">
<pre>
<code>
[1] 45
</code></pre>
</div>
```

That does shrink the after-chunk spacing slightly, but it also wrecks the grey background formatting of the code chunks: You wind up with an appropriate amount of leading gray background, but you get almost zero at the bottom of the code chunk, and it looks quite uneven.

I'm keeping my fingers crossed for a CSS fix!

**Comment by jacob-gg** [comment]
Addressed by e13b760
